### PR TITLE
[BB-197] misc pylint fixes

### PR DESCRIPTION
**Pull request by dupainaulevain**
Followup of https://github.com/open-craft/opencraft/pull/403

### CODE REVIEW

**Comment by dupainaulevain** [comment]
It turns out **unidiomatic-typecheck** suggests a modification that changes the behavior although it claims it wont. The corresponding commit was modified to just disable this misleading check and the commit message explains in detail why at **pylint: disable unidiomatic-typecheck**

File: instance/models/utils.py [hunk 1]
```diff
@@ -182,7 +182,7 @@ def __eq__(self, obj):
         To compare with inherited classes etc., use a full isinstance or issubclass expression.
         """
         if inspect.isclass(obj):
-            return type(self) is obj  # pylint: disable=unidiomatic-typecheck
+            return type(self) is obj
```
Comment (viadanna) [inline instance/models/utils.py]: This isn't idiomatic because of Python's idiosyncrasies :slightly_smiling_face:

Comment (dupainaulevain) [reply instance/models/utils.py]: That's a fine tautology, I'll keep this in mind :smile:

**Comment by dupainaulevain** [comment]
Rebased & repushed because I'm not allowed to merge unless the branch is up to date with master.

**Comment by dupainaulevain** [comment]
Rebased & repushed because I'm not allowed to merge unless the branch is up to date with master, even if there are no conflicts.

**Comment by dupainaulevain** [comment]
https://circleci.com/gh/open-craft/opencraft/5204?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-build-link probably has an environmental failure

```
======================================================================
FAIL: test_external_databases (instance.tests.integration.integration_instance.InstanceIntegrationTestCase)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/circleci/project/venv/lib/python3.5/site-packages/django/test/utils.py", line 205, in inner
    return func(*args, **kwargs)
  File "/home/circleci/project/instance/tests/integration/integration_instance.py", line 368, in test_external_databases
    self.assert_instance_up(instance)
  File "/home/circleci/project/instance/tests/integration/integration_instance.py", line 71, in assert_instance_up
    self.assertEqual(len(active_appservers), 1)
AssertionError: 0 != 1
```

**Pull request merged**